src/transactions/TransactionFrame.cpp

```diff
@@ -2024,24 +1877,345 @@ TransactionFrame::toStellarMessage() const
+{
+    auto ret = mTransactionFrame->apply(app, ltx, meta, resPayload,
+                                        sorobanBasePrngSeed);
+    mTransactionResultPayload = resPayload;
+    return ret;
+}
+
+std::pair<bool, TransactionResultPayloadPtr>
+TransactionTestFrame::checkValid(Application& app, AbstractLedgerTxn& ltxOuter,
+                                 SequenceNumber current,
+                                 uint64_t lowerBoundCloseTimeOffset,
+                                 uint64_t upperBoundCloseTimeOffset) const
+{
+    auto result = mTransactionFrame->checkValid(app, ltxOuter, current,
+                                                lowerBoundCloseTimeOffset,
+                                                upperBoundCloseTimeOffset);
+    mTransactionResultPayload = result.second;
+    return result;
+}
+
+void
+TransactionTestFrame::processFeeSeqNum(AbstractLedgerTxn& ltx,
+                                       std::optional<int64_t> baseFee)
+{
+    mTransactionResultPayload =
+        mTransactionFrame->processFeeSeqNum(ltx, baseFee);
+}
+
+void
+TransactionTestFrame::processPostApply(Application& app, AbstractLedgerTxn& ltx,
+                                       TransactionMetaFrame& meta)
+{
+    mTransactionFrame->processPostApply(app, ltx, meta,
+                                        mTransactionResultPayload);
+}
+
+bool
+TransactionTestFrame::checkValidForTesting(Application& app,
+                                           AbstractLedgerTxn& ltxOuter,
+                                           SequenceNumber current,
+                                           uint64_t lowerBoundCloseTimeOffset,
+                                           uint64_t upperBoundCloseTimeOffset)
+{
+    bool res;
+    std::tie(res, mTransactionResultPayload) =
+        checkValid(app, ltxOuter, current, lowerBoundCloseTimeOffset,
+                   upperBoundCloseTimeOffset);
+    return res;
+}
+
+bool
+TransactionTestFrame::checkSorobanResourceAndSetError(
+    Application& app, uint32_t ledgerVersion,
+    TransactionResultPayloadPtr resPayload) const
+{
+    auto ret = mTransactionFrame->checkSorobanResourceAndSetError(
+        app, ledgerVersion, resPayload);
+    mTransactionResultPayload = resPayload;
+    return ret;
+}
+
+TransactionEnvelope const&
+TransactionTestFrame::getEnvelope() const
+{
+    return mTransactionFrame->getEnvelope();
+}
+
+TransactionEnvelope&
+TransactionTestFrame::getMutableEnvelope() const
+{
+    return mTransactionFrame->getMutableEnvelope();
+}
+
+TransactionFrame const&
+TransactionTestFrame::getRawTransactionFrame() const
+{
+    auto ret =
+        std::dynamic_pointer_cast<TransactionFrame const>(mTransactionFrame);
+    releaseAssertOrThrow(ret);
+    return *ret;
+}
+
+TransactionFrameBasePtr
+TransactionTestFrame::getTxFramePtr() const
+{
+    return mTransactionFrame;
+}
+
+int64_t
+TransactionTestFrame::getFullFee() const
+{
+    return mTransactionFrame->getFullFee();
```

Comment:
I wonder, have you considered inheriting `TransactionTestFrame` directly from `TransactionFrame`? That would avoid most of the boilerplate here.